src/main/java/de/tum/cit/aet/artemis/communication/repository/FaqRepository.java

```diff
@@ -31,10 +31,18 @@ public interface FaqRepository extends ArtemisJpaRepository<Faq, Long> {
             """)
     Set<String> findAllCategoriesByCourseId(@Param("courseId") Long courseId);
 
+    @Query("""
+            SELECT DISTINCT faq.categories
+            FROM Faq faq
+            WHERE faq.course.id = :courseId AND faq.faqState = :faqState
+            """)
+    Set<String> findAllCategoriesByCourseIdAndState(@Param("courseId") Long courseId, @Param("faqState") FaqState faqState);
+
     Set<Faq> findAllByCourseIdAndFaqState(Long courseId, FaqState faqState);
 
     @Transactional
     @Modifying
     void deleteAllByCourseId(Long courseId);
 
+    Faq findByIdAndFaqState(Long faqId, Faq faqState);
```

Comment:
I am unsure what this line does, as it does not show up anywhere else. Also should the faqState be of type "Faq" or "FaqState"?

Reply:
i removed the unused code 👍